src/plugins/visualize/public/application/components/visualize_byvalue_editor.tsx

```diff
@@ -52,6 +53,17 @@ export const VisualizeByValueEditor = ({ onAppLeave }: VisualizeAppProps) => {
   const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
   const [embeddableId, setEmbeddableId] = useState<string>();
   const [valueInput, setValueInput] = useState<VisualizeInput>();
+  const { chrome, uiSettings } = services;
+  const showActionsInGroup = uiSettings.get('home:useNewHomePage');
+  const { setHeaderVariant } = chrome;
+
+  useEffect(() => {
+    if (showActionsInGroup) setHeaderVariant?.(HeaderVariant.APPLICATION);
+
+    return () => {
+      setHeaderVariant?.();
+    };
```

Comment:
If you are doing this on most of the pages, you can actually set the default header while registering the app to use APPLICATION and then set the variant to page on the page that needs it.

Reply:
I tried the method you suggested, there are some issues arising during navigations i.e when I change across the listing page and the individual editor pages.